java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -414,7 +414,15 @@ static void readRanges(FSDataInputStream file,
     long offset = first.getOffset();
     int readSize = (int) (computeEnd(first, last) - offset);
     byte[] buffer = new byte[readSize];
-    file.readFully(offset, buffer, 0, buffer.length);
+    try {
+      file.readFully(offset, buffer, 0, buffer.length);
+    } catch(IOException e) {
+      throw new IOException(String.format("Failed while reading %s %d:%d",
+                                          file,
```

Comment:
is `file.toString` only giving some object reference? I am assuming you are trying to print the path instead

Reply:
Actually, especially on S3 it prints a bunch of other data, which is helpful. I haven't looked at DFSInputStream recently, does it just use the Object.toString?

Reply:
Argh, yeah, I can see that it doesn't.

Reply:
Could you address @autumnust 's comment?